Lib/asyncio/tasks.py

```diff
@@ -408,19 +408,32 @@ async def wait(fs, *, timeout=None, return_when=ALL_COMPLETED):
     when the timeout occurs are returned in the second set.
     """
     if futures.isfuture(fs) or coroutines.iscoroutine(fs):
-        raise TypeError(f"expect a list of futures, not {type(fs).__name__}")
-    if not fs:
-        raise ValueError('Set of Tasks/Futures is empty.')
+        raise TypeError(f"expected an iterable of futures, not {type(fs).__name__}")
     if return_when not in (FIRST_COMPLETED, FIRST_EXCEPTION, ALL_COMPLETED):
         raise ValueError(f'Invalid return_when value: {return_when}')
 
-    fs = set(fs)
+    loop = events.get_running_loop()
+    new_fs = set()
 
-    if any(coroutines.iscoroutine(f) for f in fs):
-        raise TypeError("Passing coroutines is forbidden, use tasks explicitly.")
+    for f in fs:
+        if coroutines.iscoroutine(f):
+            raise TypeError("Passing coroutines is forbidden, use tasks explicitly.")
+        if not futures.isfuture(f):
+            warnings.warn(
+                "The explicit passing of awaitable objects that are not futures "
+                "to asyncio.wait() is deprecated since Python 3.11, and "
+                "scheduled for removal in Python 3.14.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
+            new_fs.add(ensure_future(f, loop=loop))
+        else:
+            new_fs.add(f)
```

Comment:
this is getting added to the set with no checks whatsoever. It needs to check `_asyncio_future_blocking`

Reply:
fixed!